screens/ExpensesScreen.tsx

```diff
@@ -11,97 +12,69 @@ import {
     >
-      <Text
-        style={{
-          color: '#40394a',
-          paddingHorizontal: 10,
-          fontWeight: '100',
-          fontSize: 30,
-          paddingVertical: 20,
-        }}
-      >
+      <Text style={styles.seeDetailsText}>
         {`See details >>>`}
       </Text>
     </View>
   );
 };
-const rightSwipeActions = () => {
+const rightSwipeActions = ({ id }: { id: number | null | undefined }) => {
   return (
-    <View
-      style={{
-        backgroundColor: '#fc0303',
-        justifyContent: 'center',
-        alignItems: 'flex-end',
-      }}
-    >
-      <Text
-        style={{
-          color: '#1b1a17',
-          paddingHorizontal: 10,
-          fontWeight: '600',
-          paddingVertical: 20,
-        }}
-      >
+    <View style={styles.deleteContainer}>
+      <Text style={styles.deleteText}>
         Delete
       </Text>
     </View>
   );
 };
-const swipeFromLeftOpen = () => {
-  alert('Swipe from left');
-};
-const swipeFromRightOpen = () => {
-  // alert('Swipe from right');
-  //do nothing
+const swipeFromLeftOpen = ({ id, navigateCallBack, swipeableRef }
+  : { id: number | null | undefined, navigateCallBack: Function, swipeableRef: React.MutableRefObject<any> }) => {
+  if (swipeableRef.current != null) {
+    const casted = swipeableRef as React.MutableRefObject<Swipeable>
+    casted.current.close()
+  }
+  navigateCallBack(id)
 };
-const ListItem = ({ id, title, amount, description, category }:
+const ListItem = ({ id, title, amount, description, category, navigateCallBack }:
   {
     id?: number | null | undefined,
     title?: String | null | undefined,
     amount?: Number | null | undefined,
     description?: String | null | undefined,
-    category?: Category | null | undefined
+    category?: Category | null | undefined,
+    navigateCallBack: Function
   }) => {
-
+  const swipeableRef = useRef(null);
   const selectedColor = (category?.colourHex) ? '#' + category.colourHex : '#03c2fc'
   return (<Swipeable
+    ref={swipeableRef}
     renderLeftActions={() => LeftSwipeActions(selectedColor)}
-    renderRightActions={rightSwipeActions}
-    onSwipeableRightOpen={swipeFromRightOpen}
-    onSwipeableLeftOpen={swipeFromLeftOpen}
+    renderRightActions={() => rightSwipeActions({ id })}
+    onSwipeableLeftOpen={() => swipeFromLeftOpen({ id, navigateCallBack, swipeableRef })}
   >
-    <View style={{
-      flex: 1,
-      flexDirection: "row",
-    }}>
+    <View style={styles.expenseItemWrapper}>
       <View style={{ flexBasis: 10, width: 10, backgroundColor: selectedColor }}></View>
-      <View
-        style={{
-          flex: 1,
-          flexDirection: "row",
-          alignContent: "space-between",
-          paddingHorizontal: 30,
-          paddingVertical: 20,
-          backgroundColor: 'white',
-        }}
-      >
-
+      <View style={styles.expenseItemDisplayContainer}>
         <Text style={{ flex: 1, fontSize: 24 }}>
-          {title}
+          {description}
```

Comment:
Display the category name instead of the description. Refer to the Figma.

Reply:
fixed